javatests/arcs/core/storage/handle/SingletonIntegrationTest.kt

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+package arcs.core.storage.handle
+
+import arcs.core.crdt.CrdtSingleton
+import arcs.core.data.EntityType
+import arcs.core.data.FieldType
+import arcs.core.data.RawEntity
+import arcs.core.data.Schema
+import arcs.core.data.SchemaDescription
+import arcs.core.data.SchemaFields
+import arcs.core.data.SchemaName
+import arcs.core.data.SingletonType
+import arcs.core.data.util.toReferencable
+import arcs.core.storage.ExistenceCriteria
+import arcs.core.storage.StorageMode
+import arcs.core.storage.StorageProxy
+import arcs.core.storage.Store
+import arcs.core.storage.StoreOptions
+import arcs.core.storage.driver.RamDisk
+import arcs.core.storage.driver.RamDiskDriverProvider
+import arcs.core.storage.driver.RamDiskStorageKey
+import arcs.core.storage.referencemode.ReferenceModeStorageKey
+import arcs.core.util.testutil.LogRule
+import com.google.common.truth.Truth.assertThat
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.test.runBlockingTest
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+
+private typealias EntitySingletonData = CrdtSingleton.Data<RawEntity>
+private typealias EntitySingletonOp = CrdtSingleton.IOperation<RawEntity>
+private typealias EntitySingletonView = RawEntity?
+
+@Suppress("EXPERIMENTAL_API_USAGE")
+@RunWith(JUnit4::class)
+class SingletonIntegrationTest {
+    @get:Rule
+    val logRule = LogRule()
+
+    private lateinit var store: Store<EntitySingletonData, EntitySingletonOp, EntitySingletonView>
+    private lateinit var storageProxy:
+        StorageProxy<EntitySingletonData, EntitySingletonOp, EntitySingletonView>
+    private lateinit var singletonA: SingletonImpl<RawEntity>
+    private lateinit var singletonB: SingletonImpl<RawEntity>
+
+    @Before
+    fun setUp() = runBlocking {
+        RamDiskDriverProvider()
+
+        store = Store(STORE_OPTIONS)
+        storageProxy = StorageProxy(store.activate(), CrdtSingleton<RawEntity>())
+
+        singletonA = SingletonImpl("collectionA", storageProxy)
```

Comment:
minor nit: name should be `singleton*` on these?

Reply:
Lol, whoops. Copy-paste has its downsides